src/jit/codegenlegacy.cpp

```diff
@@ -19216,7 +19216,7 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     noway_assert(helperNum != CORINFO_HELP_UNDEF);
 
 #ifdef FEATURE_READYTORUN_COMPILER
-                    if (call->gtEntryPoint.addr != NULL)
+                    if (call->gtEntryPoint.addr != NULL && compiler->IsTargetAbi(CORINFO_CORERT_ABI))
```

Comment:
@jkotas I guess that legacy backend has no implementation that corresponds to ``Lowering::LowerDirectCall`` (which allows jump whose call target cannot be encoded in-place).

I also wonder why Core RT does not have such an issue.